addons/mail/models/discuss/mail_message.py

```diff
@@ -1,12 +1,27 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import models, api
 from odoo.addons.mail.tools.discuss import Store
 
 
 class MailMessage(models.Model):
     _inherit = "mail.message"
 
+    @api.model_create_multi
+    def create(self, values_list):
```

Comment:
It should be implemented using the right override of mail.message methods (globally, look at ``_get_mail_message_access``usage, based notably on _mail_post_access flag). This custom code leads "yet another way of trying to do some security" and I think we have too much of them on mail models, better try to keep code in existing methods.

Reply:
I don't understand this comment, I have tried multiple solutions, but none work. `_get_mail_message_access` doesn't trigger when you are posting a message, why would it be since you are not accessing a message?

Maybe you are referring to `_get_thread_with_access`? That also seems not to be working as intended since returning `self.browse()` when you write on a read-only thread doesn't trigger any access error.

I understand your point, but I don't see a way to do it as you wish.